src/devices/Tca9548a/Tca9548A.cs

```diff
@@ -0,0 +1,229 @@
+        public bool TryGetSelectedChannel(out Channels selectedChannel)
+        {
+            try
+            {
+                var channel = _i2CDevice.ReadByte();
+                if (Enum.IsDefined(typeof(Channels), channel))
+                {
+                    selectedChannel = (Channels)channel;
+                    return true;
+                }
+                else
+                {
+                    selectedChannel = Channels.Channel0;
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                selectedChannel = Channels.Channel0;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns all the connected devices on all channels of Multiplexer
+        /// </summary>
+        /// <returns> Dictionary having Keys as Channels and Values as IEnumerable integer type</returns>
+        public Dictionary<Channels, IEnumerable<int>> ScanAllChannelsForDeviceAddress()
+        {
+            Dictionary<Channels, IEnumerable<int>> channels = new Dictionary<Channels, IEnumerable<int>>();
+            foreach (Channels channel in DeviceChannels)
+            {
+                channels.Add(channel, ScanChannelsForDeviceAddress(channel));
+            }
+
+            return channels;
+        }
+
+        /// <summary>
+        /// Returns all the connected device on selected channel
+        /// </summary>
+        /// <param name="channel">Selected Channel on Multiplexer</param>
+        /// <returns></returns>
+        public IEnumerable<int> ScanChannelsForDeviceAddress(Channels channel)
+        {
+            List<int> addresses = new List<int>();
+            SelectChannel(channel);
+            I2cDevice i2c;
+            // First 8 I2C addresses are reserved, last one is 0x7F
+            for (int i = 8; i < 0x80; i++)
+            {
+                try
+                {
+                    if (i != _i2CDevice.ConnectionSettings.DeviceAddress)
+                    {
+                        i2c = I2cDevice.Create(new I2cConnectionSettings(_i2CDevice.ConnectionSettings.BusId, i));
+                        i2c.ReadByte();
+                        addresses.Add(i);
+                    }
+                }
+                catch (IOException)
+                {
+                    i2c = null!;
+                }
+
+            }
+
+            return addresses;
+        }
+
+        /// <summary>
+        /// Enable or Disable all Mux Channels
+        /// </summary>
+        /// <param name="MUXState">Multiplexer Channel State</param>
+        public void SwitchTCA9548AState(MUXState MUXState)
+        {
+            _i2CDevice.WriteByte(Convert.ToByte(MUXState));
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            if (_shouldDispose)
+            {
+                _i2CDevice?.Dispose();
+                _i2CDevice = null!;
+            }
```

Comment:
set `_i2CDevice` to null regardless of the value of `_shouldDispose` (using this class again should throw an exception afterwards).